include/openthread/ble_secure.h

```diff
@@ -0,0 +1,469 @@
+ * @param[in]  aContext             A pointer to arbitrary context information.
+ *
+ */
+typedef void (*otHandleBleSecureConnect)(otInstance *aInstance,
+                                         bool        aConnected,
+                                         bool        aBleConnectionOpen,
+                                         void       *aContext);
+
+/**
+ * This function pointer is called when data was received over a BLE Secure TLS connection.
+ * When TLV mode is activate, the function will be called once a complete TLV was received and the
+ * message offset points to the TLV value.
+ *
+ *
+ * @param[in]  aMessage          A pointer to the message.
+ * @param[in]  aTcatMessageType  The message type received.
+ * @param[in]  aServiceName      The name of the service the message is direced to.
+ * @param[in]  aContext          A pointer to arbitrary context information.
+ *
+ */
+typedef otHandleTcatApplicationDataReceive otHandleBleSecureReceive;
+
+/**
+ * This function starts the BLE Secure service.
+ * When TLV mode is activate, the function will be called once a complete TLV was received and the
+ * message offset points to the TLV value.
+ *
+ * @param[in]  aInstance        A pointer to an OpenThread instance.
+ * @param[in]  aConnectHandler  A pointer to a function that will be called when the connection
+ *                              state changes.
+ * @param[in]  aReceiveHandler  A pointer to a function that will be called once data has been received
+ *                              over the TLS connection.
+ * @param[in]  aTlvMode         A boolean value indicating if line mode shall be activated.
+ * @param[in]  aContext         A pointer to arbitrary context information. May be NULL if not used.
+ *
+ * @retval OT_ERROR_NONE        Successfully started the BLE Secure server.
+ * @retval OT_ERROR_ALREADY     The service was stated already.
+ *
+ */
+otError otBleSecureStart(otInstance              *aInstance,
+                         otHandleBleSecureConnect aConnectHandler,
+                         otHandleBleSecureReceive aReceiveHandler,
+                         bool                     aTlvMode,
+                         void                    *aContext);
+
+/**
+ * Enables the TCAT protocol over BLE Secure.
+ *
+ * @param[in]  aInstance         A pointer to an OpenThread instance.
+ * @param[in]  aVendorInfo       A pointer to the Vendor Information (must remain valid after the method call, may be
+ * NULL).
+ * @param[in]  aHandler          A pointer to a function that is called when the join operation completes.
+ *
+ * @retval OT_ERROR_NONE              Successfully started the BLE Secure Joiner role.
+ * @retval OT_ERROR_INVALID_ARGS      @p aElevationPsk or @p aVendorInfo is invalid.
+ * @retval OT_ERROR_INVALID_STATE     The BLE function has not been started or line mode is not selected.
+ *
+ */
+otError otBleSecureTcatStart(otInstance *aInstance, otTcatVendorInfo *aVendorInfo, otHandleTcatJoin aHandler);
```

Comment:
MeshCoP::TcatAgent encapsulates the TCAT protocol and state. It may be reused by multiple link layers which could coexist. Therefore the class Ble::BleSecure has its own instance of MeshCoP::TcatAgent. Likewise another like layer technology may have its own instance of MeshCoP::TcatAgent. Therefore, it makes sense that the class Ble::BleSecure invokes TcatAgent::Start(...) inside otBleSecureTcatStart(...). 
If we were to add another TCAT link layer, such as Thread or Uart, we would start the respective TcatAgent instance with otMyOtherLinkSecureTcatStart(...).

An otTcat API certainly would make sense if we were to allow the C API user to create its own MeshCoP::TcatAgent instances for use with its own custom carrier Implementation. This is not easy because TcatAgent::Connected(...) expects a MeshCoP::Dtls reference as an abstraction of the secure carrier.  MeshCoP::Dtls is currently not exposing a C API either.
tcat.h only exposes enums and callback potentially shared between different TCAT carriers.  I suggest not adding a full otTcat API,  unless the user community badly requests custom TCAT carriers. 
I hope that makes sense.